modules/fundamental/src/vulnerability/model/details/vulnerability_advisory.rs

```diff
@@ -431,56 +439,246 @@ impl VulnerabilityAdvisoryStatus {
+                        described_by: vec![],
+                    },
+                    version: sbom_details.sbom_version,
+                    cpe: sbom_cpe,
+                    status: Default::default(),
+                }
+            } else {
+                SbomStatus {
+                    head: SbomSummary {
+                        id: sbom_id,
+                        hashes: vec![],
+                        document_id: "".to_string(),
+                        labels: Default::default(),
+                        name: "".to_string(),
+                        published: None,
+                        authors: vec![],
+                        described_by: vec![],
+                    },
+                    version: None,
+                    cpe: None,
+                    status: Default::default(),
+                }
+            };
+
+            sboms.insert(sbom_id, sbom_status);
+        }
+
+        'status: for advisory_status in sbom_purl_status {
+            if let Some(sbom_status) = sboms.get_mut(&advisory_status.sbom_id) {
+                match (advisory_status.cpe_id, &sbom_status.cpe) {
+                    (Some(_advisory_cpe), Some(sbom_cpe)) => {
+                        let mut builder = Uri::builder();
+                        if let Some(part) = advisory_status.part.as_ref() {
+                            if part != "*" {
+                                builder.part(part);
+                            }
+                        }
+                        if let Some(vendor) = advisory_status.vendor.as_ref() {
+                            if vendor != "*" {
+                                builder.vendor(vendor);
+                            }
+                        }
+                        if let Some(product) = advisory_status.product.as_ref() {
+                            if product != "*" {
+                                builder.product(product);
+                            }
+                        }
+                        if let Some(version) = advisory_status.version.as_ref() {
+                            if version != "*" {
+                                builder.version(version);
+                            }
+                        }
+                        if let Some(update) = advisory_status.update.as_ref() {
+                            if update != "*" {
+                                builder.update(update);
+                            }
+                        }
+                        if let Some(edition) = advisory_status.edition.as_ref() {
+                            if edition != "*" {
+                                builder.edition(edition);
+                            }
+                        }
+                        if let Some(language) = advisory_status.language.as_ref() {
+                            if language != "*" {
+                                builder.language(language);
+                            }
+                        }
```

Comment:
I've seen this code before, I think in this very PR. Can we DRY it up?

Reply:
I think it's this one: https://github.com/trustification/trustify/blob/cf8372a94ea5742967ac8e6262248509e2ed6f18/entity/src/cpe.rs#L152-L174